### ISSUE TITLE
Ouija Base Connector API Design

### ISSUE THREAD

**Issue by mattcreager** [ouija-io/ouija-connector-base#1]
This is a proposal for the base connector API (`ouija-connector-base` v0.1.0)

_See issue [Ouija/#76](https://github.com/ouija-io/ouija/issues/76) for context._

The base connector will wrap the **Ouija Server** API, and is responsible for retrieval, persistence and authentication, that is all.

_Note:  I'm scratching together a draft of the base connector (which will soon reside here) in the [server repo](https://github.com/ouija-io/ouija-server), simply because it's convenient._
#### Authentication and authorization

The Ouija Server will issue a valid (JWT) JSON Web Token: 
1. As the result of an OAuth authentication flow (initially with support for Twitter) - yielding a token with a scope of `member`, or in the case of the administrator `admin`
2. As the result of using the `/auth/register` endpoint - yielding a token with a scope of `guest`
#### Connection

``` es6
const BaseConnecter = require('ouija-connector-base')

// base implements connector interface, token is optional - it will request a guest token by default
let client = new BaseConnecter('https://ouija-server.herokuapp.com', {  token: 'g6x5kk..elkdqwv4' })  

client.connect()
```
#### Fetch comments for thread w. ID/Key of 1

``` es6
const BaseConnecter = require('ouija-connector-base')

let client = new BaseConnecter('https://ouija-server.herokuapp.com', {  token: 'g6x5kk..elkdqwv4' })    // base implements connector interface

client.connect()
client.get('threads/1/comments').then(comments => {
  //  [ { key: 1, body: 'comment body', author: { .. } .. } ]  --.> not sure what this looks like yet
})
```
#### Or threads, you get it.

``` es6
client.get('threads').then(threads => {
  //  [ { key: 1, name: 'sweet-blog-post' }, { key: 1, name: 'sweet-blog-post' }  ]
})
```
#### Subscribe to additions/changes to comments

``` es6
client.subscribe('thread/1/comments', (comments) =>  {
  // comments will reflect current state
})

client.subscriptions()
client.unsubscribe(path)
```
#### Disconnect

``` es6
client.disconnect({ unsubscribe: false })    // default
```
#### Add event handlers

``` es6
client.on('error', handler)   // connect, disconnect, update
```

Basically, it's [Nes](https://github.com/hapijs/nes/blob/master/API.md#client) - with sugar on top.
